types/vue.d.ts

```diff
@@ -67,8 +68,51 @@ export declare class Vue {
   $nextTick(callback: (this: this) => void): void;
   $nextTick(): Promise<void>;
   $createElement: CreateElement;
+}
+
+export type CombinedVueInstance<Data, Methods, Computed, Props> = Data & Methods & Computed & Props & Vue;
+export type ExtendedVue<Constructor extends VueConstructor, Data, Methods, Computed, Props> =
+  (new (...args: any[]) => CombinedVueInstance<Data, Methods, Computed, Props>) &
```

Comment:
Oh, I didn't understand about the mixin constructor correctly. Thank you for the explanation.